### PR TITLE
14964 user without edit permission able to update work package attributes

### PR DESCRIPTION
**Pull request by hschink**
[`* `#14964 User without edit permission able to update work package attributes`](https://www.openproject.org/work_packages/14964)

:exclamation: Consider #1766 too

### CODE REVIEW

**Comment by mfrister** [comment]
I'd say a security fix is anything but a feature.

Also, shouldn't this go into release/3.0?

**Comment by ulferts** [comment]
@meeee is right, it needs to go into dev and release/3.0

File: app/services/update_work_package_service.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  end
+
+  private
+
+  def configure_update_notification
+    JournalObserver.instance.send_notification = send_notifications
+  end
+
+  def effective_params
+    p = HashWithIndifferentAccess.new
```
Comment (myabc) [inline app/services/update_work_package_service.rb]: @hschink I would advise against using `p` as a variable name. Even though [`Kernel.p`](http://www.ruby-doc.org/core-2.1.2/Kernel.html#method-i-p) is a method and not a reserved word, creating a variable with the same name makes for ambiguity.

**Comment by mfrister** [comment]
#1766 got this into release/3.0, which will be merged back into dev, so we can close this one.

**Pull request closed without merging**